virttest/remote.py

```diff
@@ -1351,3 +1351,114 @@ def run(self, command, timeout=60, ignore_status=False):
         if status and (not ignore_status):
             raise process.CmdError(command, cmd_result)
         return cmd_result
+
+
+class VMManager(object):
+    """Manage VM on remote host"""
+
+    def __init__(self, params):
+        self.remote_host = params.get("server_ip")
+        self.remote_user = params.get("server_user")
+        self.remote_pwd = params.get("server_pwd")
+        self.runner = RemoteRunner(host=self.remote_host,
+                                   username=self.remote_user,
+                                   password=self.remote_pwd)
+
+    def setup_ssh_auth(self, vm_ip, vm_pwd, vm_user="root",
+                       port=22, timeout=20):
+        """
+        Setup SSH passwordless access between remote host
+        and VM, which is on the remote host.
+        """
+        pri_key = '~/.ssh/id_rsa'
+        pub_key = '~/.ssh/id_rsa.pub'
+        # Check the private key and public key file on remote host.
+        cmd = "ls %s %s" % (pri_key, pub_key)
+        result = self.runner.run(cmd, ignore_status=True)
+        if result.exit_status:
+            logging.debug("Create new SSH key pair")
+            self.runner.run("ssh-keygen -t rsa -q -N '' -f %s" % pri_key)
+        else:
+            logging.info("SSH key pair already exist")
+        session = self.runner.session
+        # To avoid the host key checking
+        ssh_options = "%s %s" % ("-o UserKnownHostsFile=/dev/null",
+                                 "-o StrictHostKeyChecking=no")
+        session.sendline("ssh-copy-id %s -i %s root@%s"
+                         % (ssh_options, pub_key, vm_ip))
+        while True:
```

Comment:
handle_prompts() method will be work for this below login handle right?

Reply:
@santwana Originally I did not consider to change the logic of the codes, but only simply move it.  Now I do not care to do some simple changes. So I accept your suggestion and did updates and test one case using setup_ssh_auth().

Reply:
@santwana, I am sorry that my colleague did not know  I was waiting for your reply and he merged the pr.  If you still have comments, feel free to add in this pr and I would like to consider in another pr. Thanks.